examples/js/loaders/GLTFLoader.js

```diff
@@ -2940,36 +2940,23 @@ THREE.GLTFLoader = ( function () {
 
 				var mesh = dependencies.meshes[ nodeDef.mesh ];
 
-				node = mesh.clone();
-
-				// for Specular-Glossiness
-				if ( mesh.isGroup === true ) {
+				if ( meshReferences[ nodeDef.mesh ] > 1 ) {
 
-					for ( var i = 0, il = mesh.children.length; i < il; i ++ ) {
+					node = mesh.clone();
+					node.name += '_instance_' + meshUses[ nodeDef.mesh ] ++;
 
-						var child = mesh.children[ i ];
+					// for Specular-Glossiness
+					node.onBeforeRender = mesh.onBeforeRender;
 
-						if ( child.material && child.material.isGLTFSpecularGlossinessMaterial === true ) {
+					for ( var i = 0, il = node.children.length; i < il; i ++ ) {
 
-							node.children[ i ].onBeforeRender = child.onBeforeRender;
-
-						}
+						node.children[ i ].onBeforeRender = mesh.children[ i ].onBeforeRender;
```

Comment:
Probably we should append `_instance_<N>` to the child names here, too.

Reply:
Good catch, thanks.